src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -1136,6 +1139,115 @@ export const GlobalProvider = memo(
 
         const [connectingFrom, setConnectingFrom] = useState<OnConnectStartParams | null>(null);
 
+        const getNodesBoundingBox = useCallback(() => {
+            const nodes = getNodes().filter((n) => !n.parentNode);
+
+            if (nodes.length === 0) return;
+
+            const minX = Math.min(...nodes.map((n) => n.position.x));
+            const minY = Math.min(...nodes.map((n) => n.position.y));
+            const maxX = Math.max(...nodes.map((n) => n.position.x + (n.width ?? 0)));
+            const maxY = Math.max(...nodes.map((n) => n.position.y + (n.height ?? 0)));
+
+            return {
+                x: minX,
+                y: minY,
+                width: maxX - minX,
+                height: maxY - minY,
+            };
+        }, [getNodes]);
+
+        const downloadImage = (dataUrl: string, fileName: string) => {
```

Comment:
Where will the image be downloaded to? The user's download folder?

Reply:
Currently it opens the save to dialog with a default location of the users download folder.

Reply:
I think that's good

Reply:
Same. Sounds good.